File: remove_bg.py
from PIL import Image
import os
import numpy as np

def remove_white_background(image_path):
    try:
        img = Image.open(image_path).convert("RGBA")
        data = np.array(img)
        
        # Define white threshold (e.g., pixels brighter than 240 in all channels)
        r, g, b, a = data.T
        white_areas = (r > 240) & (g > 240) & (b > 240)
        
        # Set alpha to 0 for white areas
        data[..., 3][white_areas.T] = 0
        
        # Save back
        new_img = Image.fromarray(data)
        new_img.save(image_path)
        print(f"Processed {image_path}")
    except Exception as e:
        print(f"Error processing {image_path}: {e}")

directory = "client/public/cities"
for filename in os.listdir(directory):
    if filename.endswith(".png"):
        remove_white_background(os.path.join(directory, filename))

File: client/remove_bg.mjs
import { Jimp } from 'jimp';
import fs from 'fs';
import path from 'path';

const directory = 'public/cities';

// If named import fails, try default
// import Jimp from 'jimp'; 

async function processImages() {
    try {
        const files = fs.readdirSync(directory);
        for (const file of files) {
            if (path.extname(file) === '.png') {
                const filePath = path.join(directory, file);
                console.log(`Processing ${file}...`);

                try {
                    const image = await Jimp.read(filePath);

                    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
                        const r = this.bitmap.data[idx + 0];
                        const g = this.bitmap.data[idx + 1];
                        const b = this.bitmap.data[idx + 2];

                        // If white (or close to white)
                        if (r > 240 && g > 240 && b > 240) {
                            this.bitmap.data[idx + 3] = 0; // Set alpha to 0
                        }
                    });

                    await image.write(filePath);
                    console.log(`Processed ${file}`);
                } catch (err) {
                    console.error(`Error processing ${file}:`, err);
                }
            }
        }
    } catch (err) {
        console.error('Error reading directory:', err);
    }
}

processImages();
